acceptance_test.go

```diff
@@ -0,0 +1,143 @@
+	"encoding/hex"
+	"fmt"
+	"os"
+	"sync/atomic"
+	"testing"
+
+	"github.com/conduitio-labs/conduit-connector-clickhouse/config"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/google/uuid"
+	"github.com/jmoiron/sqlx"
+	"github.com/matryer/is"
+)
+
+type driver struct {
+	sdk.ConfigurableAcceptanceTestDriver
+
+	counter int32
+}
+
+// GenerateRecord generates a random sdk.Record.
+func (d *driver) GenerateRecord(t *testing.T, operation sdk.Operation) sdk.Record {
+	atomic.AddInt32(&d.counter, 1)
+
+	return sdk.Record{
+		Position:  nil,
+		Operation: operation,
+		Metadata: map[string]string{
+			"clickhouse.table": d.Config.SourceConfig[config.Table],
+		},
+		Key: sdk.StructuredData{
+			"int_type": d.counter,
+		},
+		Payload: sdk.Change{After: sdk.RawData(
+			fmt.Sprintf(`{"int_type":%d,"string_type":"%s"}`, d.counter, uuid.NewString()),
+		)},
+	}
+}
+
+func TestAcceptance(t *testing.T) {
+	cfg := prepareConfig(t)
+
+	is := is.New(t)
+
+	sdk.AcceptanceTest(t, &driver{
+		ConfigurableAcceptanceTestDriver: sdk.ConfigurableAcceptanceTestDriver{
+			Config: sdk.ConfigurableAcceptanceTestDriverConfig{
+				Connector:         Connector,
+				SourceConfig:      cfg,
+				DestinationConfig: cfg,
+				BeforeTest: func(t *testing.T) {
+					err := createTable(cfg[config.URL], cfg[config.Table])
+					is.NoErr(err)
+				},
+				AfterTest: func(t *testing.T) {
+					err := dropTables(cfg[config.URL], cfg[config.Table])
+					is.NoErr(err)
+				},
+			},
+		},
+	})
+}
+
+// prepareConfig receives the connection URL from the environment variable
+// and prepares configuration map.
+func prepareConfig(t *testing.T) map[string]string {
+	url := os.Getenv("CLICKHOUSE_URL")
+	if url == "" {
+		t.Skip("CLICKHOUSE_URL env var must be set")
+
+		return nil
+	}
+
+	return map[string]string{
+		config.URL:            url,
+		config.Table:          fmt.Sprintf("CONDUIT_TEST_%s", randString(6)),
+		config.KeyColumns:     "int_type",
+		config.OrderingColumn: "int_type",
+	}
+}
+
+// createTable creates test table.
+func createTable(url, table string) error {
+	db, err := sqlx.Open("clickhouse", url)
+	if err != nil {
+		return fmt.Errorf("open connection: %w", err)
+	}
+	defer db.Close()
+
+	_, err = db.Exec(fmt.Sprintf(`
+	CREATE TABLE %s
+	(
+		int_type Int32,
+		string_type String
+	) ENGINE ReplacingMergeTree() PRIMARY KEY int_type;`, table))
+	if err != nil {
+		return fmt.Errorf("execute create table query: %w", err)
+	}
+
+	return nil
+}
+
+// dropTables drops test table and tracking test table if exists.
+func dropTables(url, table string) error {
+	db, err := sqlx.Open("clickhouse", url)
+	if err != nil {
+		return fmt.Errorf("open connection: %w", err)
+	}
+	defer db.Close()
+
+	_, err = db.Exec(fmt.Sprintf("DROP TABLE %s", table))
+	if err != nil {
+		return fmt.Errorf("execute drop table query: %w", err)
+	}
+
+	return nil
+}
+
+// randString generates a random string of length n.
+// (source: https://stackoverflow.com/a/47676287)
```

Comment:
why we need have link to stackoverflow ?

Reply:
I think it's not important, but I've found the link to the source in the [conduit-connector-sdk](https://github.com/ConduitIO/conduit-connector-sdk/blob/main/acceptance_testing.go#L318).

Reply:
hm, ok
But it is for you,you can leave it and close comment

Reply:
Got it. 

And you know what? I think it would be better to delete such links.

Thank you!

Reply:
Removed.